Makefile

```diff
@@ -98,36 +95,14 @@ lint-full: ## Run slower linters to detect possible issues
 .PHONY: generate
 generate: ## Generate code
 	$(MAKE) generate-manifests
-	$(MAKE) generate-go
+	$(MAKE) generate-deepcopy
 	$(MAKE) gazelle
 
-.PHONY: generate-full
-generate-full: vendor ## Generate code
-	$(MAKE) generate-clientset
-	$(MAKE) generate
-
-.PHONY: generate-go
-generate-go: ## Runs go generate
-	go generate ./pkg/... ./cmd/...
-
-.PHONY: generate-clientset
-generate-clientset: clean-clientset bin/client-gen bin/lister-gen bin/informer-gen ## Generate a typed clientset
-	bin/client-gen \
-		--clientset-name clientset \
-		--input-base sigs.k8s.io/cluster-api/pkg/apis \
-		--input deprecated/v1alpha1,cluster/v1alpha2 \
-		--output-package sigs.k8s.io/cluster-api/pkg/client/clientset_generated \
-		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
-	bin/lister-gen \
-		--input-dirs sigs.k8s.io/cluster-api/pkg/apis/deprecated/v1alpha1,sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2 \
-		--output-package sigs.k8s.io/cluster-api/pkg/client/listers_generated \
-		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
-	bin/informer-gen \
-		--input-dirs sigs.k8s.io/cluster-api/pkg/apis/deprecated/v1alpha1,sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2 \
-		--versioned-clientset-package sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset \
-		--listers-package sigs.k8s.io/cluster-api/pkg/client/listers_generated \
-		--output-package sigs.k8s.io/cluster-api/pkg/client/informers_generated \
-		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
+.PHONY: generate-deepcopy
+generate-deepcopy: ## Runs controller-gen to generate deepcopy files
+	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go \
```

Comment:
Should we handle this similar to how we've been handling controller-gen? https://github.com/kubernetes-sigs/cluster-api-provider-docker/blob/master/Makefile#L78-L80

Reply:
Definitely! I'll follow-up once we move CAPA to having a hack submodule as well